### PR TITLE
rename GooseTaskSet to Scenario, GooseTask to Transaction

### PR DESCRIPTION
**Pull request by jeremyandrews**
- rename `GooseTaskSet` to `Scenario` (and variations)
-  rename `GooseTask` to `Transaction` (and variations)
- closes https://github.com/tag1consulting/goose/issues/434
- **API change**: rename `GooseTaskSet` -> `Scenario`, `GooseTask` -> `Transaction`, `GooseTaskResult` -> `TransationResult`, `GooseTaskEror` -> `TransactionError`, `WeightedGooseTasks` -> `WeightedTransactions`, `GooseTaskFunction` -> `TransactionFunction`, `test_start_task` -> `test_start_transaction`, `test_stop_task` -> `test_stop_transaction`, `register_task` -> `register_transaction`, `task!` -> `transaction!`, `--no-task-metrics` -> `--no-transaction-metrics`, `GooseTaskError` -> `TransactionError`

### CODE REVIEW

File: src/docs/goose-book/src/getting-started/runtime-options.md [hunk 1]
```diff
@@ -5,68 +5,69 @@ The `-h` flag will show all run-time configuration options available to Goose lo
...
+Runtime options available when launching a Goose load test.
 
 
 Optional arguments:
-  -h, --help                 Displays this help
-  -V, --version              Prints version information
-  -l, --list                 Lists all tasks and exits
+  -h, --help                  Displays this help
+  -V, --version               Prints version information
+  -l, --list                  Lists all transactions and exits
```
Comment (slashrsm) [inline src/docs/goose-book/src/getting-started/runtime-options.md:14]: Same as above

**Review by slashrsm — COMMENTED** [review]
I reviewed this and did a test run. It looks OK. 

In Umami's example README file there are still few references to "tasks". Otherwise looks good to me.

File: src/config.rs [hunk 2]
```diff
@@ -35,68 +35,69 @@ const DEFAULT_PORT: &str = "5115";
...
+/// Runtime options available when launching a Goose load test.
 ///
 ///
 /// Optional arguments:
-/// -h, --help                 Displays this help
-/// -V, --version              Prints version information
-/// -l, --list                 Lists all tasks and exits
+/// -h, --help                  Displays this help
+/// -V, --version               Prints version information
+/// -l, --list                  Lists all transactions and exits
```
[the remark this answers is not in the document]
Comment (jeremyandrews) [reply src/config.rs:44]: This is auto-generated by running `--help`, I'm not sure why the spacing changed but that's from the Gumdrop library. (So yes, it's intentional)

**Comment by jeremyandrews** [comment]
@slashrsm Thanks for the review and testing! I've updated the README, hopefully this and the complimentary Glossary PR are now ready!

The tests seem to be failing because of the github cache, the code it's complaining about doesn't exist in this branch -- any ideas how to fix?

**Comment by slashrsm** [comment]
Done in #451

**Comment by jeremyandrews** [comment]
Goose Eggs is going to need an update too.

**Comment by kazimir-malevich** [comment]
Do we need to figure out how we can automatically flag API changes for both repos?

**Comment by jeremyandrews** [comment]
We increase the major version (ie, from 0.15 to 0.16) when there's an API change which is how we flag this breakage and prevent dependent code from breaking automatically. (Generally dependencies will automatically upgrade from 0.15.1 to 0.15.2 etc, just not from 0.15.2 -> 0.16).

Once the 0.16 Goose release is completed (it's getting close) then we'll need to roll a new version of Goose Eggs that's compatible.

**Pull request closed without merging**